jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/sam/sync/SyncServerlessApplicationDialog.kt

```diff
@@ -0,0 +1,444 @@
+import software.amazon.awssdk.services.cloudformation.model.StackSummary
+import software.amazon.awssdk.services.cloudformation.model.Tag
+import software.amazon.awssdk.services.ecr.EcrClient
+import software.amazon.awssdk.services.lambda.model.PackageType
+import software.amazon.awssdk.services.s3.S3Client
+import software.aws.toolkits.jetbrains.core.awsClient
+import software.aws.toolkits.jetbrains.core.map
+import software.aws.toolkits.jetbrains.services.cloudformation.CloudFormationTemplate
+import software.aws.toolkits.jetbrains.services.cloudformation.Parameter
+import software.aws.toolkits.jetbrains.services.cloudformation.SamFunction
+import software.aws.toolkits.jetbrains.services.cloudformation.describeStack
+import software.aws.toolkits.jetbrains.services.cloudformation.mergeRemoteParameters
+import software.aws.toolkits.jetbrains.services.cloudformation.resources.CloudFormationResources
+import software.aws.toolkits.jetbrains.services.ecr.CreateEcrRepoDialog
+import software.aws.toolkits.jetbrains.services.ecr.resources.EcrResources
+import software.aws.toolkits.jetbrains.services.ecr.resources.Repository
+import software.aws.toolkits.jetbrains.services.lambda.deploy.CapabilitiesEnumCheckBoxes
+import software.aws.toolkits.jetbrains.services.lambda.deploy.CreateCapabilities
+import software.aws.toolkits.jetbrains.services.lambda.sam.SamTemplateUtils
+import software.aws.toolkits.jetbrains.services.lambda.sam.ValidateSamParameters.validateParameters
+import software.aws.toolkits.jetbrains.services.lambda.sam.ValidateSamParameters.validateStackName
+import software.aws.toolkits.jetbrains.services.s3.CreateS3BucketDialog
+import software.aws.toolkits.jetbrains.services.s3.resources.S3Resources
+import software.aws.toolkits.jetbrains.settings.SyncSettings
+import software.aws.toolkits.jetbrains.settings.relativeSamPath
+import software.aws.toolkits.jetbrains.ui.KeyValueTextField
+import software.aws.toolkits.jetbrains.ui.ResourceSelector
+import software.aws.toolkits.jetbrains.utils.ui.validationInfo
+import software.aws.toolkits.jetbrains.utils.ui.withBinding
+import software.aws.toolkits.resources.message
+import java.awt.Component
+
+data class SyncServerlessApplicationSettings(
+    val stackName: String,
+    val bucket: String,
+    val ecrRepo: String?,
+    val parameters: Map<String, String>,
+    val tags: Map<String, String>,
+    val useContainer: Boolean,
+    val capabilities: List<CreateCapabilities>
+)
+
+class SyncServerlessApplicationDialog(
+    private val project: Project,
+    private val templateFile: VirtualFile,
+    private val loadResourcesOnCreate: Boolean = true
+) : DialogWrapper(project) {
+    var useContainer: Boolean = false
+    var newStackName: String = ""
+    var syncType: SyncType = SyncType.CREATE
+    var createNewStack = true
+    var templateParameters: Map<String, String> = emptyMap()
+    var tags: Map<String, String> = emptyMap()
+    var showImageOptions: Boolean = false
```

Comment:
instead of declaring this twice, why not create an instance of `SyncServerlessApplicationSettings` and store the state directly in that instance?